Modules/EMCAL/src/ClusterTask.cxx

```diff
@@ -113,36 +137,59 @@ void ClusterTask::startOfCycle()
+    std::vector<int> CellIndex;
+    std::vector<o2::emcal::TriggerRecord> ClusterTR, CellIndexTR;
 
-  InputCluster = "emcal-clusters";
-  InputClusterTrig = "emcal-clustertriggerecords";
+    findClustersInternal(Cell, CellTR, Cluster, ClusterTR, CellIndex, CellIndexTR); //do internal clustering here
+    ILOG(Info, Support) << "Found  " << Cluster.size() << " CLusters  " << ENDM;
+    ILOG(Info, Support) << "Found  " << ClusterTR.size() << " ClusterTR  " << ENDM;
+    ILOG(Info, Support) << "Found  " << CellIndex.size() << " CellIndex  " << ENDM;
+    ILOG(Info, Support) << "Found  " << CellIndexTR.size() << " CellIndexTR  " << ENDM;
 
-  InputCellIndex = "emcal-cellindices";
-  InputCellIndexTrig = "emcal-citriggerecords";
+    analyseTimeframe(Cell, CellTR, Cluster, ClusterTR, CellIndex, CellIndexTR); // Fill histos
+  } else {
 
-  auto Cell = ctx.inputs().get<gsl::span<o2::emcal::Cell>>(InputCell.c_str());
+    std::string InputCluster = "emcal-clusters";
+    std::string InputClusterTR = "emcal-clustertriggerecords";
 
-  auto CellTriggerRecord = ctx.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(InputCellTrig.c_str());
+    std::string InputCellIndex = "emcal-cellindices";
+    std::string InputCellIndexTR = "emcal-citriggerecords";
 
-  auto Cluster = ctx.inputs().get<gsl::span<o2::emcal::Cluster>>(InputCluster.c_str());
+    auto Cluster = ctx.inputs().get<gsl::span<o2::emcal::Cluster>>(InputCluster.c_str());
+    auto ClusterTR = ctx.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(InputClusterTR.c_str());
+    auto CellIndex = ctx.inputs().get<gsl::span<int>>(InputCellIndex.c_str());
+    auto CellIndexTR = ctx.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(InputCellIndexTR.c_str());
 
-  auto ClusterTriggerRecord = ctx.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(InputClusterTrig.c_str());
+    analyseTimeframe(Cell, CellTR, Cluster, ClusterTR, CellIndex, CellIndexTR); // Fill histos
+  }
+}
 
-  auto CellIndex = ctx.inputs().get<gsl::span<int>>(InputCellIndex.c_str());
+void ClusterTask::endOfCycle()
+{
+  ILOG(Info, Support) << "endOfCycle" << ENDM;
+}
 
-  auto CellIndexTriggerRecord = ctx.inputs().get<gsl::span<o2::emcal::TriggerRecord>>(InputCellIndexTrig.c_str());
+void ClusterTask::endOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info, Support) << "endOfActivity" << ENDM;
+}
 
+//_____________________________  Fill function _____________________________
+void ClusterTask::analyseTimeframe(const gsl::span<const o2::emcal::Cell>& cells, const gsl::span<const o2::emcal::TriggerRecord>& cellTriggerRecords, const gsl::span<const o2::emcal::Cluster> clusters, const gsl::span<const o2::emcal::TriggerRecord> clusterTriggerRecords, const gsl::span<const int> clusterIndices, const gsl::span<const o2::emcal::TriggerRecord> cellIndexTriggerRecords)
+{
   mEventHandler = std::make_unique<o2::emcal::EventHandler<o2::emcal::Cell>>();
   mClusterFactory = std::make_unique<o2::emcal::ClusterFactory<o2::emcal::Cell>>();
```

Comment:
The objects are already created and configured in initialize, you must not reassign them here. When reassigning a std::unique_ptr it means that the object which was assigned to it is deleted. This is not only a resource waste, it also means you loose the configuration. In particular since we now need to propagate the geometry to the cluster factory it will not be visible in the version of the object you create here.

You only need to reset the objects via
``` c++
  mEventHandler->reset();
  mClusterFactory->reset();
```

Reply:
I have removed the lines where I was reassigning both the objects and I reset them at beginning of analyseTimeframe function.